v1/daemon/write.go

```diff
@@ -15,19 +15,53 @@
 package daemon
 
 import (
-	"fmt"
+	"context"
+	"io"
+	"io/ioutil"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
 
 	"github.com/google/go-containerregistry/name"
 	"github.com/google/go-containerregistry/v1"
+	"github.com/google/go-containerregistry/v1/tarball"
 )
 
+// API interface for testing.
+type ImageLoader interface {
+	ImageLoad(context.Context, io.Reader, bool) (types.ImageLoadResponse, error)
+}
+
+// This is a variable so we can override in tests.
+var getImageLoader = func() (ImageLoader, error) {
+	return client.NewEnvClient()
+}
+
 // WriteOptions are used to expose optional information to guide or
 // control the image write.
 type WriteOptions struct {
 	// TODO(dlorenc): What kinds of knobs does the daemon expose?
 }
 
 // Write saves the image into the daemon as the given reference.
-func Write(ref name.Reference, img v1.Image, wo WriteOptions) error {
-	return fmt.Errorf("NYI: daemon.Write(%v)", ref)
+func Write(ref name.Reference, img v1.Image, wo WriteOptions) (string, error) {
+	cli, err := getImageLoader()
+	if err != nil {
+		return "", err
+	}
+
+	tag, err := name.NewTag(ref.Name(), name.WeakValidation)
+	if err != nil {
+		return "", err
+	}
+	// write the image in docker save format first, then load it
+	reader := tarball.Write(tag, img, &tarball.WriteOptions{})
+	resp, err := cli.ImageLoad(context.Background(), reader, false)
+	b, err := ioutil.ReadAll(resp.Body)
```

Comment:
It might be better to just make `daemon.Write` return an `io.ReadCloser` (i.e. `resp.Body`) instead of a string? I'm not sure how this will be used, so it's hard to say.